### ISSUE TITLE
iPad editor renders double cursor

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#121195]
iPad, safari

1. Open settings UI
2. Touch some non input area, for example the Settings table of content
3. Touch back on the settings input area -> double cursor is rendered

Apart from this steps sometimes in various editors I see double cursors rendered.

I was able to fix this by adding
`-webkit-user-select: none;` to the hidden `textArea` [here](https://github.com/microsoft/vscode/blob/main/src/vs/editor/browser/controller/textAreaHandler.css#L17)

I am not sure if that would break anything else, thus first creating this issue to discuss if you think this is a good aproach or if you are already aware of this issue

fyi @alexdima

**Comment by alexdima** [comment]
@isidorn Did you push that change? Adding `-webkit-user-select: none;` to the editor's input `textarea` might break IME (like typing Chinese, etc) so we would need to test that or add that only for the iPad.

**Comment by isidorn** [comment]
@alexdima I did not push that change since I was curious to hear your opinion.
I suggest that I push this change this week in debt week and I try it out with IME chinese? What do you think?

**Comment by alexdima** [comment]
Double blinking cursors indicate that the `<textarea>` is made visible. But the `<textarea>` is normally made visible only when doing IME (i.e. composition) so I would personally first try to find out why is the `<textarea>` made visible, is there some composition event that triggers that, etc. 

So my first attempt at fixing would be to figure out why is the `<textarea>` shown and suppress that if it is not necessary. Only if we cannot deal with this, would I go to tweak the CSS to try to hide the cursor in the visible `<textarea>`.

**Comment by isidorn** [comment]
@alexdima thanks for the pointer. However if I put a breakpoint there it does not get hit. So that code does not get executed - we do not have a composition. Is there some other scenario when `<textarea>` can be made visible?

Inspecting the elements here's what I see:
So the width of the `textArea` is `1px`. If I set that manually to 0px in dev tools the cursor is no longer shown, however that probably defeats the purpose since also in VS Code desktop I see `1px` width of the `textArea`.

![Screenshot 2021-06-07 at 20 43 39](https://user-images.githubusercontent.com/1926584/121072175-25ba7100-c7d1-11eb-9ccb-d731e02d82a2.png)

**Comment by alexdima** [comment]
@isidorn Did you find an easy way to run this in an emulator on macOS? Regarding the 1px width of the `<textarea>`. On Chromium, having a 0x0 `<textarea>` completely breaks user input, like typing in it does not generate any events, I think it cannot be focused properly by Chrome.

Does typing work on the iPad when relaxing [this constant](https://github.com/microsoft/vscode/blob/11a1f705db94886bd3a33c2d547dc35699998a34/src/vs/editor/browser/controller/textAreaHandler.ts#L57) to be `const canUseZeroSizeTextarea = (browser.isFirefox || browser.isSafari);`. If this works fine on the iPad then I can also help with testing on Safari.

**Comment by isidorn** [comment]
@alexdima no, I did not find a good way to emulate. I know @misolori uses an emulator, I am not sure if it is only for design layouts or is it the full thing. Maybe he can provide more details.

When I relax that constant to be `browser.isFirefox || browser.isSafari` the typing works just fine on the iPad and this bug is no longer reproducable.

**Comment by misolori** [comment]
You can use Xcode to open the simulator, which will give you access to all iOS devices/platforms, and then use Safari on your desktop to connect to the dev tools if you need to inspect.

**Comment by isidorn** [comment]
Moving this to next milestone when Alex Dima is back from vacation and then hopefully we can investigate into this together.

**Comment by isidorn** [comment]
Now that we have an iPad in the office, it might be interesting to set it up and look into this?
Assigning also @hediet and @alexdima, if you do not have time feel free to push out of September.

**Comment by 1shaked** [comment]
Where can I download the Ipad version???

**Comment by petercall** [comment]
Hi guys, did you ever find a way to fix this issue?
When I code on Google Colab on my iPad, my iPad is displaying 2 cursors.
I will attach a picture. The black cursor is the actual cursor for the page, and then the blue cursor is a ghost cursor that has no editing power, but that hovers near the black cursor and makes it very visually difficult to code because of the distraction of 2 cursors. 

Let me know if you figured out how to fix this.
Thank you!

![IMG_0071](https://github.com/microsoft/vscode/assets/106123313/4865c464-8565-4813-b592-e856e24025b6)
